### PR TITLE
changes to make MinGW work

### PR DESCRIPTION
**Pull request by Alexander12365**
Changes to make MinGW builds work. Some preprocessor directives in src files and some if-else in cmake checking if compiler/architecture is MinGW.

https://stackoverflow.com/questions/71712556/mariadb-connector-cpp-1-0-1-linking-error-for-cjportedtests-exe

### CODE REVIEW

File: src/CArrayImp.h [hunk 1]
```diff
@@ -56,11 +56,14 @@ template <class T> CArray<T>::CArray(int64_t len, const T& fillValue) : CArray<T
   std::fill(this->begin(), this->end(), fillValue);
 }
 
-#ifndef _WIN32
+#if defined(__MINGW32__) || defined(__MINGW64__)
```
Comment (lawrinn) [inline src/CArrayImp.h:59]: Here would be simpler to make one #ifdef _WIN32 and the rest, it seems

**Comment by lawrinn** [comment]
Thank you for making a pull request, as I asked you for, and sorry for getting back to you with such a long delay. Atm I am working on the new release, and would like to merge the PR's I have. 
Unfortunately I need from you a couple of more things. First - we need you to state explicitly, that you are contributing your new code of the whole pull request, including one or several files that are either new files or modified ones, under the BSD-new license.
Second, there are conflicts, and would be great if you could resolve them. But that I can actually do myself. But first is a must - I can't merge without that.

**Comment by Alexander12365** [comment]
Hey lawrinn,

I will glady do so. I hope this comment satisfies the request.

I hereby want to state, that the code of the whole pull request, including any new or modified files, falls hereby under the BSD-new license.

**Comment by lawrinn** [comment]
Thanks! Ok, no problem - I'll resolve the conflicts. I will need to  brush the code a bit anyway

**Comment by lawrinn** [comment]
I did some my changes - hopefully they won't break the build

**Pull request merged**